### PR TITLE
[tools] wallet lib tests and cleanup

### PR DESCRIPTION
**Pull request by saki-osive**
In this PR, the following have been addressed

- Pass the remaining tests within the wallet directory
- Clean up the code
- Add doc comments to improve readability
- Add a minimal README file

### CODE REVIEW

**Review by dboreham — APPROVED** [review]

File: tools/wallet/src/core/legacy_scheme.rs [hunk 1]
```diff
@@ -42,3 +42,36 @@ impl LegacyKeyScheme {
         LegacyKeyScheme::new(&wallet)
     }
 }
+
+#[cfg(test)]
```
Comment (0o-de-lally) [inline tools/wallet/src/core/legacy_scheme.rs:46]: I don't think this needs to be created as a `mod`. You can just move this to a test section, and import `use diem_crypto::Length;` with a cfg(tests) annotation.

File: tools/wallet/src/core/mnemonic.rs [hunk 2]
```diff
@@ -6,10 +6,9 @@
 //!
 //! https://github.com/rust-bitcoin/rust-wallet/blob/master/wallet/src/mnemonic.rs
 
-// use crate::error::WalletError;
 use anyhow::{bail, Result};
-// #[cfg(test)]
-// use diem_temppath::TempPath;
+#[cfg(test)]
```
Comment (0o-de-lally) [inline tools/wallet/src/core/mnemonic.rs:10]: let's cleanup the ordering of these imports please. Also for all files in this module. There are cargo tools for reordering imports.

File: tools/wallet/src/core/wallet_library.rs [hunk 3]
```diff
@@ -12,8 +12,6 @@
 //! https://github.com/rust-bitcoin/rust-wallet/blob/master/wallet/src/walletlibrary.rs
 
 use super::{
```
Comment (0o-de-lally) [inline tools/wallet/src/core/wallet_library.rs:14]: I'm unclear if this is the idiomatic way to do these imports. Please check.

File: tools/wallet/src/core/wallet_library.rs [hunk 4]
```diff
@@ -151,3 +149,88 @@ impl WalletLibrary {
         &self.key_factory
     }
 }
+
+#[cfg(test)]
```
Comment (0o-de-lally) [inline tools/wallet/src/core/wallet_library.rs:153]: same comment as elsewhere, this doesn't need to be a `mod`

File: tools/wallet/src/key_gen.rs [hunk 5]
```diff
@@ -21,6 +20,16 @@ pub fn keygen() -> (AuthenticationKey, AccountAddress, WalletLibrary, String) {
     (auth_key, account, wallet, mnemonic_string)
 }
 
+/// Runs the key generation process asynchronously.
```
Comment (0o-de-lally) [inline tools/wallet/src/key_gen.rs:23]: I do like this pattern of being explicit with the parameter documentation.
Though in general we don't see that pattern in rust.
This is a more idiomatic way of writing comments: https://www.tangramvision.com/blog/making-great-docs-with-rustdoc

**Review by 0o-de-lally — CHANGES_REQUESTED** [review]
In general this is an excellent cleanup! Left a few comments on some Rust conventions. Great stuff!

File: tools/wallet/src/core/legacy_scheme.rs (hunk 1 above)
Comment (saki-osive) [reply tools/wallet/src/core/legacy_scheme.rs]: The test has been updated.

File: tools/wallet/src/core/mnemonic.rs (hunk 2 above)
Comment (saki-osive) [reply tools/wallet/src/core/mnemonic.rs]: Reorganized all imports in the entire module.

File: tools/wallet/src/core/wallet_library.rs (hunk 3 above)
Comment (saki-osive) [reply tools/wallet/src/core/wallet_library.rs:14]: Reorganized all imports in the entire module.

File: tools/wallet/src/core/wallet_library.rs (hunk 4 above)
Comment (saki-osive) [reply tools/wallet/src/core/wallet_library.rs]: Updated

File: tools/wallet/src/key_gen.rs (hunk 5 above)
Comment (saki-osive) [reply tools/wallet/src/key_gen.rs]: Updated doc.

**Comment by 0o-de-lally** [comment]
Looks good to me, I'm just waiting to see if we can get the tests passing again before merging.

**Review by 0o-de-lally — APPROVED** [review]
LGTM!

**Comment by 0o-de-lally** [comment]
duplicated in types cleanup PR

**Pull request closed without merging**